src/cfi.rs

```diff
@@ -63,6 +63,78 @@ impl<'input, Endian> DebugFrame<'input, Endian>
         let (_, entry) = try!(CommonInformationEntry::parse(input));
         Ok(entry)
     }
+
+    /// Find the frame unwind information for the given address.
+    ///
+    /// If found, the unwind information is returned along with the reset
+    /// context in the form `Ok((unwind_info, context))`. If not found,
+    /// `Err(gimli::Error::NoUnwindInfoForAddress)` is returned. If parsing or
+    /// CFI evaluation fails, the error is returned.
+    ///
+    /// ```
+    /// use gimli::{DebugFrame, NativeEndian, UninitializedUnwindContext};
+    ///
+    /// # fn foo() -> gimli::Result<()> {
+    /// # let read_debug_frame_section = || unimplemented!();
+    /// // Get the `.debug_frame` section from the object file.
+    /// let debug_frame = DebugFrame::<NativeEndian>::new(read_debug_frame_section());
+    ///
+    /// # let get_frame_pc = || unimplemented!();
+    /// // Get the address of the PC for a frame you'd like to unwind.
+    /// let address = get_frame_pc();
+    ///
+    /// // This context is reusable, which cuts down on heap allocations.
+    /// let ctx = UninitializedUnwindContext::new();
+    ///
+    /// let (unwind_info, ctx) = try!(debug_frame.unwind_info_for_address(ctx, address));
```

Comment:
Does the borrow checker allow the following syntax? I'm worried all the moves are going to hurt; we can reevaluate once there is benchmark.
```
let mut ctx = UninitializedUnwindContext::new();
let unwind_info = try!(debug_frame.unwind_info_for_address(&mut ctx, address));
```
Also, I like that you're using `try!` in this example. We should update the other examples to do this.